### ISSUE TITLE
Cannot set build name/number with mvn

### ISSUE THREAD

**Issue by jvalkeal** [jfrog/jfrog-cli#537]
Running
```
jfrog rt mvn "clean install -Dartifactory.publish.buildInfo=true" --build-name=mybuild --build-number=1
```
always pushes build name taken from pom file and build number as timestamp.

Looking tmp file i.e. `buildInfoProperties771564722`  passed to build contains correct info:
```
buildInfoConfig.includeEnvVars=false
artifactory.publish.contextUrl=http://localhost:8081/artifactory
artifactory.deploy.build.name=mybuild
buildInfo.generated.build.info=/tmp/jfrog/builds/bXlidWlsZF8x/generatedBuildInfo280541289
buildInfoConfig.envVarsExcludePatterns=*password*,*secret*,*key*,*token*
buildInfo.agent.name=jfrog-cli-go/1.32.2
artifactory.deploy.build.timestamp=1576746141408
buildInfo.licenseControl.autoDiscover=true
buildInfo.licenseControl.includePublishedArtifacts=false
artifactory.deploy.build.number=1
artifactory.publish.buildInfo=false
buildInfo.licenseControl.runChecks=false
artifactory.publish.artifacts=true
org.jfrog.build.extractor.maven.recorder.activate=true
artifactory.publish.repoKey=libs-release
artifactory.publish.snapshot.repoKey=libs-snapshot
```
But `target/build-info.json` contains:
```
  "version" : "1.0.1",
  "name" : "demo",
  "number" : "1576759545421",
  "type" : "MAVEN",
  "buildAgent" : {
    "name" : "Maven",
    "version" : "3.5.3"
  },
  "agent" : {
    "name" : "jfrog-cli-go/1.32.2",
    "version" : "3.5.3"
  },
```

**Comment by jvalkeal** [comment]
Btw almost feels like #398 and its fix but it was done just a gradle.

**Comment by eyalbe4** [comment]
@jvalkeal,
I believe the issue was resolved on the latest patch released today.
Try this with version 1.32.3 and let me know if it works for you.

**Comment by jvalkeal** [comment]
@eyalbe4 nope, didn't see it working any better with that version.

**Comment by eyalbe4** [comment]
@jvalkeal,
I took a closer look at the command you shared above. I'm not sure actually if this worked in previous versions, but being able to publish the build info without the **jfrog rt bp** command has never been tested or meant to be supported. 
Please try the following with the latest JFrog CLI release:
1. Configure your maven project's repositories by running the **jfrog rt mvnc** while in the root dir of the project. The configuration will be stored in a *.jfrog* directory at the root of the project. 
2. Run the mvn build ss follows (no quotes or the *-Dartifactory.publish.buildInfo=true* system properly):
**jfrog rt mvn clean install --build-name=mybuild --build-number=1**
3. Then, publish the build info as follows:
**jfrog rt bp my-build 1**

Please let me know if this works for you.

**Comment by jvalkeal** [comment]
I looked from sources how go interacts with build info and it really looks 

```
artifactory.deploy.build.name=mybuild
artifactory.deploy.build.number=1
```
Should possibly be
```
buildInfo.build.name=mybuild
buildInfo.build.number=1
```
unless in maven things are handled differently. This was a fix for gradle in 9e13a195680801fdd0fae51b3e58e14284f464ac.

Also it's in docs that things should work like I used it.

You need to have `-Dartifactory.publish.buildInfo=true` as it's false by default:
```
$ jfrog rt mvn clean install --build-name=mybuild --build-number=1

[main] INFO org.jfrog.build.extractor.maven.BuildDeploymentHelper - Artifactory Build Info Recorder: Saving Build Info to '/home/jvalkealahti/repos/jvalkeal/atest1/target/build-info.json'
[main] INFO org.jfrog.build.extractor.maven.BuildInfoClientBuilder - Deploying artifact: http://localhost:8081/artifactory/libs-snapshot/com/example/demo/0.0.1.BUILD-SNAPSHOT/demo-0.0.1.BUILD-SNAPSHOT.jar
[main] INFO org.jfrog.build.extractor.maven.BuildInfoClientBuilder - Deploying artifact: http://localhost:8081/artifactory/libs-snapshot/com/example/demo/0.0.1.BUILD-SNAPSHOT/demo-0.0.1.BUILD-SNAPSHOT.pom
[main] INFO org.jfrog.build.extractor.maven.BuildDeploymentHelper - Artifactory Build Info Recorder: publish build info set to false, build info will not be published...
```
This command really does nothing.
```
$ jfrog rt bp my-build 1
[Debug] Sending usage info...
[Debug] Sending HTTP GET request to: http://localhost:8081/artifactory/api/system/version
[Debug] The Artifactory version is: 6.16.0
[Debug] Sending HTTP POST request to: http://localhost:8081/artifactory/api/system/usage
[Debug] Artifactory response: 200 OK
[Debug] Usage info sent successfully.
[Error] open /tmp/jfrog/builds/bXktYnVpbGRfMQ==/partials/details: no such file or directory
```

**Comment by jvalkeal** [comment]
Not sure if there were some fixes in `1.32.4` but now build info lands into artifactory ok. I took a simple start.spring.io project for gradle/maven.

```
$ jfrog rt mvn --build-name=mymaven --build-number=m1 clean install
$ jfrog rt bp mymaven m1
$ jfrog rt gradle --build-name=mygradle --build-number=g1 clean build artifactoryPublish
$ jfrog rt bp mygradle g1
```

I think what goes wrong with maven and `-Dartifactory.publish.buildInfo=true` is that local `build-info.json` contains wrong `name` and `number`. However it looks like when you use `jfrog rt bp` with maven and gradle actual build info in artifactory looks ok.

Build Info JSON in artifactory from maven build:
```
{
    "version": "1.0.1",
    "name": "mymaven",
    "number": "m1",
```

Build Info JSON in artifactory from gradle build:
```
{
    "version": "1.0.1",
    "name": "mygradle",
    "number": "g1",
```

It's just that the local `build-info.json` for maven is wrong which put me in a wrong track.
```
$ head -n 10 target/build-info.json 
{
  "version" : "1.0.1",
  "name" : "demoartifactory",
  "number" : "1577628666151",
  "type" : "MAVEN",
  "buildAgent" : {
    "name" : "Maven",
    "version" : "3.6.3"
  },
  "agent" : {

14:14 $ head -n 10 build/build-info.json 
{
  "version" : "1.0.1",
  "name" : "mygradle",
  "number" : "g1",
  "type" : "GRADLE",
  "buildAgent" : {
    "name" : "Gradle",
    "version" : "6.0.1"
  },
  "agent" : {
```

**Comment by eyalbe4** [comment]
@jvalkeal,
I believe this issue can be closed.
Please let me know if you'd like to discuss this further.
